adding changelog for Wrangler interactive improvement for Assets deployment (#25012)
cloudflare/cloudflare-docs: src/content/changelog/workers/2025-09-09-interactive-wrangler-assets.mdx

---

> title: Interactive deployment flow for static assets in Wrangler

Reviewer: ```suggestion
title: Deploy static sites to Workers without a configuration file
```

Reviewer: (we have habit of calling this `static assets` but really what we're talking about / what people are familar with is "static sites" — when a site has no backend, as is the case here)

description: Wrangler now guides you through deploying static assets with interactive prompts and automatic configuration generation when no wrangler.jsonc file exists.
products:
  - workers
date: 2025-09-09
---

> Deploying static assets to Workers is now significantly easier. When you run `wrangler deploy [directory]` or `wrangler deploy --assets [directory]` without an existing configuration file, Wrangler will now guide you through the deployment process with interactive prompts.

Reviewer: ```suggestion
Deploying static site to Workers is now easier. When you run `wrangler deploy [directory]` or `wrangler deploy --assets [directory]` without an existing [configuration file](/workers/wrangler/configuration/), [Wrangler CLI](/workers/wrangler/) now guides you through the deployment process with interactive prompts.
```

## What's new

**Interactive prompts for missing configuration:**
- Wrangler detects when you're trying to deploy a directory of static assets
- Prompts you to confirm the deployment type
- Asks for a project name (with smart defaults)
- Automatically sets the compatibility date to today

**Automatic configuration generation:**
- Creates a `wrangler.jsonc` file with your deployment settings
- Stores your choices for future deployments
- Eliminates the need to remember complex command-line flags

## Example workflow

```bash
# Deploy your built static site

> wrangler deploy dist

Reviewer: Is the best example `wrangler deploy dist` or just `wrangler deploy`? Felt like the latter is what really shows that we figure this out for you?

# Wrangler will prompt:
✔ It looks like you are trying to deploy a directory of static assets only. Is this correct? … yes
✔ What do you want to name your project? … my-astro-site

> # Automatically generates wrangler.jsonc:

Reviewer: ```suggestion
# Automatically generates a wrangler.jsonc file and adds it to your project:
```

{
  "name": "my-astro-site",
  "compatibility_date": "2025-09-09",
  "assets": {
    "directory": "dist"
  }
}

> # Future deployments are now simplified:

Reviewer: ```suggestion
# Next time you run wrangler deploy, this will use the configuration in your newly generated wrangler.jsonc file
```

wrangler deploy
```

## Before and after

> **Before:** Required remembering multiple flags and parameters

Reviewer: I would bring this up here way at the top after first paragraph, line 10

```bash
wrangler deploy --assets ./dist --compatibility-date 2025-09-09 --name my-project
```

**After:** Simple directory deployment with guided setup
```bash
wrangler deploy dist
# Interactive prompts handle the rest as shown in the example flow above
```

## Requirements

> - **Wrangler 4.24.4 or higher** is required for this functionality

Reviewer: ```suggestion
- You must use Wrangler version 4.24.4 or later in order to use this feature
```

2nd person imperative more direct and clear for docs than passive "is required"
